webapp/channels/src/components/admin_console/dropdown_setting.tsx

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import Setting from './setting';
+
+type Props = {
+    id: string;
+    values: Array<{ text: string; value: string }>;
+    label: React.ReactNode;
+    value: string;
+    onChange(id: string, value: any): void;
+    disabled?: boolean;
+    setByEnv: boolean;
+    helpText?: React.ReactNode;
+}
+const DropdownSetting = ({
+    id,
+    values,
+    label,
+    value,
+    onChange,
+    disabled = false,
+    setByEnv,
+    helpText,
+}: Props) => {
+    const handleChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+        onChange(id, e.target.value);
+    };
+
+    const options = [];
+    for (const {value, text} of values) {
+        options.push(
+            <option
+                value={value}
+                key={value}
+            >
+                {text}
+            </option>,
+        );
+    }
```

Comment:
nit : can this be just placed in return with suitable array methods? If we cant then ok with keeping it here

Reply:
Done